spinalcordtoolbox/scripts/sct_process_segmentation.py

```diff
@@ -320,27 +367,54 @@ def main(argv=None):
                                          angle_correction=angle_correction,
                                          param_centerline=param_centerline,
                                          verbose=verbose)
+    if fname_pmj is not None:
+        im_ctl, mask, slices = get_slices_for_pmj_distance(fname_segmentation, fname_pmj,
+                                                           distance_pmj, extent_mask,
+                                                           param_centerline=param_centerline,
+                                                           verbose=verbose)
+        # Save mask
+        fname_mask_out = add_suffix(arguments.i, '_mask_csa')
+        mask.save(fname_mask_out)
+        # Save extrapolated centerline
+        fname_ctl = add_suffix(arguments.i, '_centerline_extrapolated')
+        im_ctl.save(fname_ctl)
+
+        # Generated centerline smoothed in RL direction for visualization (and QC report)
+        fname_ctl_smooth = add_suffix(fname_ctl, '_smooth')
+        subprocess.run(['sct_maths', '-i', fname_ctl, '-smooth', '10,1,1', '-o', fname_ctl_smooth], stdout=subprocess.PIPE, check=True)
```

Comment:
I wanted to generate the smoothed centerline, I wasn't sure if there was another wat to call `sct_maths `for it? (I wanted to use the API` maths.py`, but I would have still repeated some lines from `sct_maths.py`)

Reply:
At the very least, you can import `sct_maths` then access it via `sct_maths.main()` to avoid the subprocess call.

Reply:
At the very least, you can import `sct_maths` then access it via `sct_maths.main()` to avoid the subprocess call.

Reply:
great i'll do that! thanks!